samza-core/src/main/scala/org/apache/samza/metrics/JvmMetrics.scala

```diff
@@ -153,4 +163,16 @@ class JvmMetrics(val registry: MetricsRegistry) extends MetricsHelper with Runna
     gThreadsTimedWaiting.set(threadsTimedWaiting)
     gThreadsTerminated.set(threadsTerminated)
   }
+
+  private def updateOperatingSystemMetrics {
+    if (osMXBean.isInstanceOf[OperatingSystemMXBean]) {
+      val operatingSystemMXBean = osMXBean.asInstanceOf[OperatingSystemMXBean]
+      gProcessCpuUsage.set(operatingSystemMXBean.getProcessCpuLoad * PCT)
+      gSystemCpuUsage.set(operatingSystemMXBean.getSystemCpuLoad * PCT)
+
+      if (osMXBean.isInstanceOf[UnixOperatingSystemMXBean]) {
+        gOpenFileDescriptorCount.set(osMXBean.asInstanceOf[UnixOperatingSystemMXBean].getOpenFileDescriptorCount)
```

Comment:
Since Samza only supports *nix systems, I wonder if we can directly check for `UnixOperatingSystemMxBean` (this extends from `OperatingSystemMXBean` anyways?) 
It should work for posix systems like *nix, and os-x.

Reply:
Spoken like a non-MSFT employee :-)

I prefer it this way, just in case future versions of the JVM start returning a different type. It's frustrating that we have to do these checks and casts in a virtual machine, but since we do, I'd like them to be as robust as possible.